conf/SimpleConfiguration.php

```diff
@@ -0,0 +1,268 @@
+			if(is_array($properties)) {
+				foreach($properties as $property => $value) {
+					if(is_array($value)) {
+						$this->dynamicConfigs[] = "this.${subconfig}.${property}";
+					}
+				}
+			}
+		}
+
+		$this->configs["base"] = &$this->configs;
+	}
+
+	protected function parseDynamicConfigs($config = null, $resolve = false)
+	{
+		if(is_null($config)) {
+			foreach($this->dynamicConfigs as $config) {
+				$val = "";
+				$expandedConfig = self::getVariableByAlias($config);
+				foreach($expandedConfig as $portion) {
+					if(is_array($portion)) {
+						$val .= $this->parseDynamicConfigs($portion, true);
+					}
+					else {
+						$val .= $portion;
+					}
+				}
+				self::setVariableByAlias($config, $val);
+			}
+		}
+		else {
+			$val = "";
+			if(is_array($config)) {
+				if(isset($config["check"])) {
+					$matches = array();
+					if(isset($config["check"][1])) {
+						if(preg_match("/{(.*)}/si", $config["check"][1], $matches)) {
+							$val1 = self::getVariableByAlias($matches[1]);
+						}
+						else {
+							$val1 = $config["check"][1];
+						}
+					}
+
+					if(isset($config["check"][2])) {
+						if(preg_match("/{(.*)}/si", $config["check"][2], $matches)) {
+							$val2 = self::getVariableByAlias($matches[2]);
+						}
+						else {
+							$val2 = $config["check"][2];
+						}
+					}
+
+					switch($config["check"][0])
+					{
+						case "=":
+							if($val1 == $val2) {
+								return $this->parseDynamicConfigs($config["true"]);
+							}
+							else {
+								return $this->parseDynamicConfigs($config["false"]);
+							}
+							break;
+						case "<>":
+							if($val1 != $val2) {
+								return $this->parseDynamicConfigs($config["true"]);
+							}
+							else {
+								return $this->parseDynamicConfigs($config["false"]);
+							}
+							break;
+						case "<":
+							if($val1 < $val2) {
+								return $this->parseDynamicConfigs($config["true"]);
+							}
+							else {
+								return $this->parseDynamicConfigs($config["false"]);
+							}
+							break;
+						case ">":
+							if($val1 > $val2) {
+								return $this->parseDynamicConfigs($config["true"]);
+							}
+							else {
+								return $this->parseDynamicConfigs($config["false"]);
+							}
+							break;
+						case "<=":
+							if($val1 <= $val2) {
+								return $this->parseDynamicConfigs($config["true"]);
+							}
+							else {
+								return $this->parseDynamicConfigs($config["false"]);
+							}
+							break;
+						case ">=":
+							if($val1 >= $val2) {
+								return $this->parseDynamicConfigs($config["true"]);
+							}
+							else {
+								return $this->parseDynamicConfigs($config["false"]);
+							}
+							break;
+					}
+				}
+				else {
+					foreach($config as $portion) {
+						if(is_array($portion)) {
+							$val .= $this->parseDynamicConfigs($portion, true);
+						}
+						else {
+							if($resolve) {
+								$portionVal = self::getVariableByAlias($portion);
+								if(is_array($portionVal)) {
```

Comment:
Should probably add a comment here indicating that this was added to fix the race condition on conditional configurations.